client/client_test.go

```diff
@@ -1795,3 +1796,18 @@ func TestClient_ReconnectAllocs(t *testing.T) {
 	require.False(t, invalid, "expected alloc to not be marked invalid")
 	require.Equal(t, unknownAlloc.AllocModifyIndex, finalAlloc.AllocModifyIndex)
 }
+
+func TestClient_DefaultIneligible(t *testing.T) {
```

Comment:
This test seems like it is attempting to match the exact conditions seen within the code which as previously noted is a bit of a red herring.